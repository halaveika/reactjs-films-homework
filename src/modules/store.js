import { createStore, applyMiddleware, combineReducers } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import contentReducer from './content';

const rootReducer = combineReducers({ content: contentReducer });

const middleware = [thunk];
const store = createStore(
  rootReducer,
  composeWithDevTools(
    applyMiddleware(...middleware),
  ),
);

export default store;
