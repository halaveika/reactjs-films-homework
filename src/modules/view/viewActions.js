import {SET_LIST_FLEX_DIRECTION,SET_GENRE } from '../../constants';

export const setGenre = (genre) => ({ type: SET_GENRE, payload: genre });

export const setListFlexDirection = (boolean) => ({ type: SET_LIST_FLEX_DIRECTION, payload: boolean });
