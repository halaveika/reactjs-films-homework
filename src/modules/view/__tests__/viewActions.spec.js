import { setGenre, setListFlexDirection } from '../viewActions';
import { store } from '../../store';

describe('view actions', () => {
  test('setGenre actioin change store', async () => {
    await store.dispatch(setGenre('Drama'));
    expect(store.getState()).toMatchSnapshot();
  });

  test('setListFlexDirection actioin change store', async () => {
    await store.dispatch(setListFlexDirection(false));
    expect(store.getState()).toMatchSnapshot();
  });
});
