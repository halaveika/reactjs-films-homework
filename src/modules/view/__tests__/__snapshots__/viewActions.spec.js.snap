// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`view actions setGenre actioin change store 1`] = `
Object {
  "content": Object {
    "details": Object {},
    "filter": "TRENDING",
    "genres_array": Array [],
    "isLoading": true,
    "page": 1,
    "pageSize": 20,
    "results": Array [],
    "searchValue": "",
    "total_pages": 0,
    "total_results": 0,
    "video_url": "",
  },
  "view": Object {
    "genre": "Drama",
    "isRow": true,
  },
}
`;

exports[`view actions setListFlexDirection actioin change store 1`] = `
Object {
  "content": Object {
    "details": Object {},
    "filter": "TRENDING",
    "genres_array": Array [],
    "isLoading": true,
    "page": 1,
    "pageSize": 20,
    "results": Array [],
    "searchValue": "",
    "total_pages": 0,
    "total_results": 0,
    "video_url": "",
  },
  "view": Object {
    "genre": "Drama",
    "isRow": false,
  },
}
`;
